### PR TITLE
Updates to network analysis

### LINKED ISSUE #411: Would be great to add expected influence 'centrality' in the network analysis module.
* Enhancement:
* Purpose:
* Use-case:

**Is your feature request related to a problem? Please describe.**
 

**Describe the solution you'd like**
 

**Describe alternatives you've considered**
 

**Additional context**

### LINKED ISSUE #702: Network analysis - Changing Legend does not change the Centrality Plot
*** JASP version:** 0.12.1

*** Analysis:** Network
*** Bug description:** This new version has advanced in-network graphic models, but the centrality plot is still having problems.
Note that on the centrality plot (shown in the legend), the system still displays the legend.
When the nodes only display the number (the name of the variable is displayed in the legend), the centrality plot still displays the variable name. Thus
Probem
, it is not possible to insert the same data in a scientific article, for example.
Please, see the images attached.

* Expected behaviour:

### LINKED ISSUE #751: Network analysis: SPELLING MISTAKE
JASP-Network-Bootstrap Options-Bootstrap Type-Nonperamet*ic

### PR DESCRIPTION
**Pull request by vandenman**
Fixes https://github.com/jasp-stats/jasp-issues/issues/411
Fixes https://github.com/jasp-stats/jasp-issues/issues/702
Fixes https://github.com/jasp-stats/jasp-issues/issues/751
Analog of https://github.com/jasp-stats/Network/pull/1

[Diff without whitespace changes](https://github.com/jasp-stats/jasp-desktop/pull/4077/files?w=1)

The changes:

- When `Show variable names in legend` is ticked, the variable names are now shown in a legend for the centrality and clustering plot.
- Facets of the centrality plot can be toggled.
- Added a new centrality measure called `expected influence`.
- Values of edges can now be shown inside the network plot
- Fixed a bug in using manual colors for coloring nodes.

### CODE REVIEW

**Comment by vandenman** [comment]
The first commit failed on Travis because the tests for the centrality table had an extra column.

**Comment by TimKDJ** [comment]
Can you rebase?

**Comment by vandenman** [comment]
@TimKDJ I think I rebased the correct commits away.. (edit: the correct commits as in the ones that should not be merged into stable)

File: JASP-Engine/JASP/R/networkanalysis.R [hunk 1]
```diff
@@ -16,32 +16,32 @@
 #
 
 NetworkAnalysis <- function(jaspResults, dataset, options) {
-
+
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/networkanalysis.R]: It's one thing to remove unnecessary spacing, but to add it? ;)

File: JASP-Engine/JASP/R/networkanalysis.R [hunk 2]
```diff
@@ -582,101 +604,107 @@ NetworkAnalysis <- function(jaspResults, dataset, options) {
...
-
+      
       groups <- vector("list", nGroups)
       names(groups) <- groupNames[, 1L]
       for (i in seq_len(nGroups))
         groups[[i]] <- which(idx == i)
-
+      
+      nonEmpty <- lengths(groups) > 0L
+      groups <- groups[nonEmpty]
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/networkanalysis.R:699]: I'm not sure where this orginates exactly, but to be sure: we're certain `groups` will always have at least one non empty entry?

**Review by TimKDJ — CHANGES_REQUESTED** [review]
There were a few problems I found with these changes:
- [x] The `Expected influence` column is empty, instead it shows a column titled `NA` next to it
- [x] I get the error `invalid 'width' argument` in the centrality plot

[bugs_network_pr.zip](https://github.com/jasp-stats/jasp-desktop/files/4716051/bugs_network_pr.zip) (rename)

File: JASP-Engine/JASP/R/networkanalysis.R (hunk 1 above)
Comment (vandenman) [reply JASP-Engine/JASP/R/networkanalysis.R]: hahaha I'm not sure how that happened but I'll get rid of it :p

File: JASP-Engine/JASP/R/networkanalysis.R (hunk 2 above)
Comment (vandenman) [reply JASP-Engine/JASP/R/networkanalysis.R:699]: `groups` is based on a QML element that cannot be empty.

**Comment by vandenman** [comment]
@TimKDJ refreshing the jasp file should fix all the errors. In addition, I made a mistake while rebasing and accidentally removed changes to the QML file (which caused the width error).

**Review by TimKDJ — APPROVED** [review]

**Pull request merged**